test_scripts/API/VehicleData/SubscribeVehicleData/004_Resumption_success_flow_2_apps.lua

```diff
@@ -0,0 +1,187 @@
+-- 2) App1 and app2 send valid SubscribeVehicleData to SDL and these requests are allowed by Policies
+-- 3) Apps reconnect
+-- SDL must:
+-- 1) Perform resumption of data
+-- 2) Transfer SubscribeVehicleData request to HMI and by resumption of app1
+-- 3) not transfer SubscribeVehicleData request to HMI and by resumption of app2
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/VehicleData/commonVehicleData')
+local actions = require("user_modules/sequences/actions")
+local test = require("user_modules/dummy_connecttest")
+local mobile_session = require("mobile_session")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+config.application1.registerAppInterfaceParams.isMediaApplication = false
+config.application1.registerAppInterfaceParams.appHMIType = { "NAVIGATION" }
+config.application2.registerAppInterfaceParams.isMediaApplication = true
+config.application2.registerAppInterfaceParams.appHMIType = { "DEFAULT" }
+
+local hashIdArray = { }
+
+local rpc = {
+  name = "SubscribeVehicleData",
+  params = {
+    engineOilLife = true,
+    fuelRange = true,
+    tirePressure = true
+  }
+}
+
+local vehicleDataResults = {
+  engineOilLife = {
+    dataType = "VEHICLEDATA_ENGINEOILLIFE",
+    resultCode = "SUCCESS"
+  },
+  fuelRange = {
+    dataType = "VEHICLEDATA_FUELRANGE",
+    resultCode = "SUCCESS"
+  },
+  tirePressure = {
+    dataType = "VEHICLEDATA_TIREPRESSURE",
+    resultCode = "SUCCESS"
+  }
+}
+
+--[[ Local Functions ]]
+local function getVDParams()
+  local out = {}
+  for k in pairs(common.allVehicleData) do
+    table.insert(out, k)
+  end
+  return out
+end
+
+local function ptUpdateForApps(pTbl)
+  pTbl.policy_table.app_policies[common.getConfigAppParams(1).fullAppID].groups = { "Base-4", "Emergency-1" }
+  pTbl.policy_table.app_policies[common.getConfigAppParams(2).fullAppID].groups = { "Base-4", "Emergency-1" }
+  local grp = pTbl.policy_table.functional_groupings["Emergency-1"]
+  for _, v in pairs(grp.rpcs) do
+    v.parameters = getVDParams()
+  end
+  pTbl.policy_table.vehicle_data = nil
+end
+
+local function processRPCSubscribeSuccess(pAppId, pHMIsubscription)
+  local mobileSession = common.getMobileSession( pAppId)
+  local cid = mobileSession:SendRPC(rpc.name, rpc.params)
+  if true == pHMIsubscription then
+    EXPECT_HMICALL("VehicleInfo." .. rpc.name, rpc.params)
+    :Do(function(_, data)
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+          vehicleDataResults)
+      end)
+  else
+    EXPECT_HMICALL("VehicleInfo." .. rpc.name)
+    :Times(0)
+  end
+  local responseParams = vehicleDataResults
+  responseParams.success = true
+  responseParams.resultCode = "SUCCESS"
+  mobileSession:ExpectResponse(cid, responseParams)
```

Comment:
```suggestion
  mobileSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS", common.vehicleDataResults })
```
Comment is related to all scripts.